### PR TITLE
Additional services exercise proposed by Jean Gareau.

### PR DESCRIPTION
**Pull request by kgleong**

### CODE REVIEW

File: app/src/main/java/com/facebook/workshop/class4bservicecoding/CounterService.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  }
+
+  @Override
+  public IBinder onBind(Intent intent) {
+    return null;
+  }
+
+  @Override
+  public void onDestroy() {
+    mServiceLooper.quit();
```
Comment (kgleong) [inline app/src/main/java/com/facebook/workshop/class4bservicecoding/CounterService.java:65]: call `quitSafely()` instead?

Comment (nesquena) [inline app/src/main/java/com/facebook/workshop/class4bservicecoding/CounterService.java:65]: Yes except this method only exists > API 18 which is a problem if people support API 16+ http://stackoverflow.com/a/20332075/313399

File: app/src/main/java/com/facebook/workshop/class4bservicecoding/MainActivity.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  TextView mCounterTextView;
+
+  /**
+   * This broadcast receiver updates the counter TextView with the
+   * CounterService.COUNT intent char sequence value.
+   */
+  BroadcastReceiver mBroadcastReceiver = new BroadcastReceiver() {
+    @Override
+    public void onReceive(Context context, Intent intent) {
+      mCounterTextView.setText(intent.getCharSequenceExtra(CounterService.COUNT));
```
Comment (kgleong) [inline app/src/main/java/com/facebook/workshop/class4bservicecoding/MainActivity.java]: possibly could leave this as a `TODO` as well

File: app/src/main/java/com/facebook/workshop/class4bservicecoding/CounterService.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+    @Override
+    public void handleMessage(Message msg) {
+      Intent intent = new Intent(ACTION);
+      intent.putExtra(COUNT, Integer.toString(++mCount));
+
+      // TODO: broadcast the message to the activity
+      mLocalBroadcastManager.sendBroadcast(intent);
+
+      mServiceHandler.sendEmptyMessageDelayed(0, DELAY_MS);
```
Comment (kgleong) [inline app/src/main/java/com/facebook/workshop/class4bservicecoding/CounterService.java:39]: maybe leave this as a TODO as this is a nice test of the class's understanding.

File: app/src/main/java/com/facebook/workshop/class4bservicecoding/CounterService.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    thread.start();
+    mServiceLooper = thread.getLooper();
+
+    // An Android service handler is a handler running on a specific background thread.
+    mServiceHandler = new ServiceHandler(mServiceLooper);
+    mServiceHandler.sendEmptyMessageDelayed(0, DELAY_MS);
+  }
+
+  @Override
+  public IBinder onBind(Intent intent) {
```
Comment (kgleong) [inline app/src/main/java/com/facebook/workshop/class4bservicecoding/CounterService.java:59]: curious: why this method is implemented?  as a discussion point?

**Comment by kgleong** [comment]
I can see this as a replacement for the broadcast receiver demo.

File: app/src/main/java/com/facebook/workshop/class4bservicecoding/CounterService.java (hunk 1 above)
Comment (kgleong) [inline app/src/main/java/com/facebook/workshop/class4bservicecoding/CounterService.java:65]: thanks makes sense :+1:

**Pull request merged**